src/DynamicContent/Strings.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace WPML\PB\Elementor\DynamicContent;
+
+use WPML\Collect\Support\Collection;
+use WPML_PB_String;
+
+class Strings {
+
+	const SETTINGS_REGEX        = '/settings="(.*?(?="]))/';
+	const NAME_PREFIX           = 'dynamic';
+	const DELIMITER             = '-';
+	const TRANSLATABLE_SETTINGS = [
+		'before',
+		'after',
+		'fallback',
+	];
+
+	/**
+	 * Remove the strings overwritten with dynamic content
+	 * and add the extra strings "before", "after" and "fallback".
+	 *
+	 * @param WPML_PB_String[] $strings
+	 * @param string            $nodeId
+	 * @param array             $element
+	 *
+	 * @return WPML_PB_String[]
+	 */
+	public static function filter( array $strings, $nodeId, array $element ) {
+
+		$dynamicFields = wpml_collect( isset( $element['settings']['__dynamic__'] ) ? $element['settings']['__dynamic__'] : [] );
+
+		$setStringKey = function ( WPML_PB_String $string ) {
+			list( $fieldName ) = explode( '-', $string->get_name() );
```

Comment:
@strategio I assume `list( $fieldName )` just gets the first element in the array. Why not:

```php
return explode( '-', $string->get_name() )[0];
```